consensus/hotstuff/cruisectl/block_time_controller_test.go

```diff
@@ -223,59 +216,89 @@ func (bs *BlockTimeControllerSuite) TestInit_EpochSetupPhase() {
-		return now.Add(bs.config.FallbackProposalDelay.Load()) == bs.ctl.getProposalTiming().TargetPublicationTime(7, now, unittest.IdentifierFixture())
+		return len(bs.ctl.epochEvents) == 0
 	}, time.Second, time.Millisecond)
 
-	// additional EpochFallbackModeTriggered events should be no-ops
-	// (send capacity+1 events to guarantee one is processed)
-	for i := 0; i <= cap(bs.ctl.epochFallbacks); i++ {
-		bs.ctl.EpochFallbackModeTriggered(0, nil)
+	extensionTargetTime, err := bs.snapshot.Epochs().Current().TargetEndTime()
+	require.NoError(bs.T(), err)
+	extensionTargetDuration, err := bs.snapshot.Epochs().Current().TargetDuration()
+	require.NoError(bs.T(), err)
+	extensionFinalView, err := bs.snapshot.Epochs().Current().FinalView()
+	require.NoError(bs.T(), err)
+
+	assert.Equal(bs.T(), extensionTargetTime, bs.ctl.curEpochTargetEndTime)
+	assert.Equal(bs.T(), extensionTargetDuration, bs.ctl.curEpochTargetDuration)
+	assert.Equal(bs.T(), extensionFinalView, bs.ctl.curEpochFinalView)
+
+	// duplicate events should be no-ops
+	for i := 0; i <= cap(bs.ctl.epochEvents); i++ {
+		bs.ctl.EpochExtended(bs.epochCounter, header, extension)
 	}
-	// state should be unchanged
-	now := time.Now().UTC()
-	assert.Equal(bs.T(), now.Add(bs.config.FallbackProposalDelay.Load()), bs.ctl.getProposalTiming().TargetPublicationTime(12, now, unittest.IdentifierFixture()))
+	require.Eventually(bs.T(), func() bool {
+		return len(bs.ctl.epochEvents) == 0
+	}, time.Second, time.Millisecond)
 
-	// additional OnBlockIncorporated events should be no-ops
-	for i := 0; i <= cap(bs.ctl.incorporatedBlocks); i++ {
-		header := unittest.BlockHeaderFixture(unittest.HeaderWithView(bs.initialView + 1 + uint64(i)))
-		header.ParentID = unittest.IdentifierFixture()
-		bs.ctl.OnBlockIncorporated(model.BlockFromFlow(header))
+	assert.Equal(bs.T(), extensionTargetTime, bs.ctl.curEpochTargetEndTime)
+	assert.Equal(bs.T(), extensionFinalView, bs.ctl.curEpochFinalView)
+}
+
+// TestOnEpochCommittedPhaseStarted ensures that the epoch info is updated when the next epoch is committed.
+func (bs *BlockTimeControllerSuite) TestOnEpochCommittedPhaseStarted() {
+	nextEpoch := mockprotocol.NewEpoch(bs.T())
+	nextEpoch.On("Counter").Return(bs.epochCounter+1, nil)
+	nextEpoch.On("FinalView").Return(bs.curEpochFinalView*2, nil)
+	nextEpoch.On("TargetDuration").Return(bs.EpochDurationSeconds(), nil)
+	nextEpoch.On("TargetEndTime").Return(bs.curEpochTargetEndTime+bs.EpochDurationSeconds(), nil)
+	bs.epochs.Add(nextEpoch)
+	bs.CreateAndStartController()
+	defer bs.StopController()
+	header := unittest.BlockHeaderFixture()
+	bs.ctl.EpochCommittedPhaseStarted(bs.epochCounter, header)
+	require.Eventually(bs.T(), func() bool {
+		return bs.ctl.nextEpochFinalView != nil
+	}, time.Second, time.Millisecond)
+
+	assert.Equal(bs.T(), bs.curEpochFinalView*2, *bs.ctl.nextEpochFinalView)
+	assert.Equal(bs.T(), bs.curEpochTargetEndTime+bs.EpochDurationSeconds(), *bs.ctl.nextEpochTargetEndTime)
```

Comment:
Could we also check that `bs.ctl.curEpochTargetDuration` has the expected value?

Reply:
https://github.com/onflow/flow-go/pull/6102/commits/c1bb33e73346e1922c5ab69916b340b971b84e93